classes/order/Order.php

```diff
@@ -2688,10 +2688,13 @@ public function refreshShippingCost()
         $this->update();
 
         // save order_carrier prices, we'll save order right after this in update() method
-        $order_carrier = new OrderCarrier((int) $this->getIdOrderCarrier());
-        $order_carrier->shipping_cost_tax_excl = $this->total_shipping_tax_excl;
-        $order_carrier->shipping_cost_tax_incl = $this->total_shipping_tax_incl;
-        $order_carrier->update();
+        $orderCarrierId = (int) $this->getIdOrderCarrier();
```

Comment:
It's fun that this method is capable of returning zero and look right in your eyes "Eyh, here's the order ID you asked for, nothing weird here, sir !" 😆